{"@class": "de.fau.cs.inf2.cthree.data.Cluster", "id": "3c00c80939dc7b290ce9188be3cc9cf6", "detectedBy": ["DIFF_DBSCAN"], "members": [{"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "3f78e90666e5ff76551af3efe3a3a9f4", "repository": "http://git.eclipse.org/gitroot/jdt/eclipse.jdt.core.git", "fileName": "org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/CompilationUnitProblemFinder.java", "commitBeforeChange": "4d3248bf8a50c2573f33ebfb354f201ec2af10f0", "commitAfterChange": "50b056417f7b2dfe8902e626cfbc3e6979160ac1", "methodNumberBeforeChange": 6, "methodNumberAfterChange": 6, "signatureBeforeChange": "  \tpublic static CompilationUnitDeclaration process( \t\tCompilationUnitDeclaration unit, \t\tICompilationUnit unitElement,  \t\tIProblemRequestor problemRequestor, \t\tIProblemFactory problemFactory, \t\tIProgressMonitor monitor) \t\tthrows JavaModelException", "signatureAfterChange": "  \tpublic static CompilationUnitDeclaration process( \t\tCompilationUnitDeclaration unit, \t\tICompilationUnit unitElement,  \t\tIProblemRequestor problemRequestor, \t\tIProblemFactory problemFactory, \t\tIProgressMonitor monitor) \t\tthrows JavaModelException", "diff": ["+\t\t} catch(RuntimeException e) { ", "+\t\t\t// avoid breaking other tools due to internal compiler failure (40334)", "+\t\t\tUtil.log(e, \"Exception occurred during problem detection: \"); //$NON-NLS-1$ ", "+\t\t\tthrow new JavaModelException(e, IJavaModelStatusConstants.COMPILER_FAILURE);"]}, {"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "f7f8135c1d3309ffbc63a13732622dff", "repository": "http://git.eclipse.org/gitroot/jdt/eclipse.jdt.core.git", "fileName": "org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/JavaProject.java", "commitBeforeChange": "77f63ea750ed0364fe2343040edb02da4f79c989", "commitAfterChange": "c7494793aa1ed6a2f75240b5a076591b5a3e9bf6", "methodNumberBeforeChange": 44, "methodNumberAfterChange": 44, "signatureBeforeChange": " \tpublic IPath getOutputLocation() throws JavaModelException", "signatureAfterChange": " \tpublic IPath getOutputLocation() throws JavaModelException", "diff": ["+\t\t} catch (Assert.AssertionFailedException e) { // failed creating CP entries from file", "+\t\t\tif (JavaModelManager.VERBOSE && this.getProject().isAccessible()){", "+\t\t\t\tUtil.log(e, ", "+\t\t\t\t\t\"Exception while retrieving \"+ this.getPath() //$NON-NLS-1$", "+\t\t\t\t\t+\"/.classpath, will revert to default output location\"); //$NON-NLS-1$", "+\t\t\t}"]}], "num": 50394}